src/state/routing/useRoutingAPITrade.ts

```diff
@@ -3,14 +3,16 @@ import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
 import { IMetric, MetricLoggerUnit, setGlobalMetric } from '@uniswap/smart-order-router'
 import { sendTiming } from 'components/analytics'
 import { AVERAGE_L1_BLOCK_TIME } from 'constants/chainInfo'
-import { useStablecoinAmountFromFiatValue } from 'hooks/useStablecoinPrice'
 import { useRoutingAPIArguments } from 'lib/hooks/routing/useRoutingAPIArguments'
 import ms from 'ms.macro'
 import { useMemo } from 'react'
 import { RouterPreference, useGetQuoteQuery } from 'state/routing/slice'
 
-import { InterfaceTrade, TradeState } from './types'
-import { computeRoutes, transformRoutesToTrade } from './utils'
+import { InterfaceTrade, QuoteState, TradeState } from './types'
+
+const TRADE_INVALID = { state: TradeState.INVALID, trade: undefined }
+const TRADE_NOT_FOUND = { state: TradeState.NO_ROUTE_FOUND, trade: undefined }
+const TRADE_LOADING = { state: TradeState.LOADING, trade: undefined }
```

Comment:
nit: Since we're returning those objects to the outside world, it might be good to add `as const` to each declaration to ensure they're truly read-only. Otherwise, we risk (very unlikely, but still) interference across different calls. It's theoretical, but still.